test/extended/networking/external_gateway.go

```diff
@@ -0,0 +1,115 @@
+package networking
+
+import (
+	"context"
+	"net"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+type Family string
+
+const (
+	IPv4      Family = "ipv4"
+	IPv6      Family = "ipv6"
+	DualStack Family = "dual"
+	Unknown   Family = "unknown"
+)
+
+var _ = g.Describe("[sig-network] external gateway address", func() {
+	oc := exutil.NewCLI("ns-global")
+
+	InOVNKubernetesContext(func() {
+		f := oc.KubeFramework()
+
+		g.It("should match the address family of the pod", func() {
+			podIPs, err := createPod(f.ClientSet, f.Namespace.Name, "test-valid-gateway-pod")
+			expectNoError(err)
+			podIPFamily := getIPFamily(podIPs)
+			// Set external gateway address into an IPv6 address and make sure
+			// pod ip address matches with IPv6 address family.
+			makeNamespaceWithExternalGatewaySet(f, "fd00:10:244:2::6")
+			podIPs, err = createPod(f.ClientSet, f.Namespace.Name, "test-ipv6-pod")
+			if err != nil {
+				o.Expect(getIPFamily(podIPs)).To(o.Equal(Unknown))
```

Comment:
So I'm guessing this is the "expected failure" case where the pod fails to get created if it has an ipv4 only addr, but the gw is set to ipv6? If so, I think a comment here would be good explicitly saying that.